### ISSUE TITLE
Neovide crashing while "working" Value not an Array: '{"kind": "syntax", "hi_name": "SignColumn"... }

### ISSUE THREAD

**Issue by lucax88x** [neovide/neovide#2452]
**Describe the bug**
This morning neovide just keep crashing because of treesitter highlight change (check the logs below)

**To Reproduce**
Checking logs, might related to redraw of neovide?

**Expected behavior**
To not crash

**Desktop (please complete the following information):**
 - OS: 14.4.1
 - Neovide Version 0.12.2
 - Neovim Version v0.10.0-dev-2739+ge2224a793

[crash log](https://gist.github.com/lucax88x/1d36ed14f4fa6657499ab4b75809b697)

**Comment by fredizzimo** [comment]
This is either some regression in Neovim nighly or some new even we don't handle. I will take a look at it soon, but do you remember when you last updated Neovim before today?

**Comment by lucax88x** [comment]
no more than 2/3 days

**Comment by fredizzimo** [comment]
Unfortunately. I can't reproduce this here, it's probably caused by some plugin, can you try to run `neovide -- --clean` and see if it happens?

**Comment by fredizzimo** [comment]
Most likely the `nvim-treesitter-context` plugin, I will try to install that here

**Comment by CertainLach** [comment]
Also experiencing with `nvim_input_mouse` when I try to enable https://github.com/lewis6991/hover.nvim

```
Caused by:
    0: Error decoding response to request 'nvim_input_mouse'
    1: Error while decoding
    2: Value not an Array: '{"kind": "syntax", "hi_name": "GruvboxRedUnderline", "id": 167}'
```

(I understand this is not a neovide issue per se)

**Comment by fredizzimo** [comment]
@CertainLach, thanks, is that plugin enough to reproduce the issue? If you follow this:
https://github.com/neovim/neovim/blob/master/contrib/minimal.lua, 

If you prefer you can alternatively create a single file `lazy.nvim` configuration that can be launched with `NVIM_APPNAME=notarray neovide`, the config then goes into `.config/notarray` instead of the normal`.config.nvim`

**Comment by lucax88x** [comment]
Really hard to reproduce sadly, it happens randomly.

**Comment by vargad** [comment]
I'm having the same issue with neovide 0.12.2, and the latest neovim v0.10.0-dev-2806+g9af355964-dirty. I'm not using treesitter nor hover, not sure which plugin triggers it.
```
ERROR [neovide::bridge::ui_commands] Mouse Scroll Failed

Caused by:
    0: Error decoding response to request 'nvim_input_mouse'
    1: Error while decoding
    2: Value not an Array: '{"kind": "syntax", "hi_name": "SpellCap", "id": 173}'
ERROR [neovide::bridge] Error: 'Error reading message'
```

**Comment by fredizzimo** [comment]
Maybe it's not a plugin. I have quite strong hunch that this might be doing something wrong 
* https://github.com/neovim/neovim/pull/27655

It seem to do so in the terminal mode:
* https://github.com/neovim/neovim/issues/27859

Would it be possible for someone to test it either by reverting https://github.com/neovim/neovim/commit/55c9e2c96e28059a5102e2551b16b4fa4592723b, or a with a version prior to it?

I don't think there are any readymade binaries available for that anymore, so you probably have to build Neovim yourself.

**Comment by vargad** [comment]
A conflict was not trivial to resolve so I made a build with https://github.com/neovim/neovim/commit/6525832a8c4d44a8ebabba02a5ea1ce09b389a4f . Will see if it stops crashing. I believe this version will likely make neovide freeze.

**Comment by vargad** [comment]
Sorry, I was a bit busy in the past days, but so far no crash, no hang. This https://github.com/neovim/neovim/commit/6525832a8c4d44a8ebabba02a5ea1ce09b389a4f neovim revision seems good with the latest neovide release.

**Comment by fredizzimo** [comment]
Thanks, I reported our findings here:
https://github.com/neovim/neovim/issues/27859#issuecomment-2044764163

**Comment by fredizzimo** [comment]
Can you all test with the latest Neovim nightly build from today? There's another potential fix here
* https://github.com/neovim/neovim/pull/28717

**Comment by fredizzimo** [comment]
The fix is most likely fixed in Neovim 0.10.0 and since our new 0.13.0 release requires that, I'm closing this issue.
